### ISSUE TITLE
Queue has 2 messages after consumption and returning 1 message the queue is now empty.

### ISSUE THREAD

**Issue by jhnferraris** [squaremo/amqp.node#279]
I experimented on putting multiple messages on a single queue `queue-1`. The messages are `message-1` and `message-2`

```
[
    message-2
    message-1
]
```

When I executed this:

```
var q = 'queue-1'
function consumer(conn) {
  var ok = conn.createChannel(on_open);
  function on_open(err, ch) {
    if (err != null) bail(err);
    ch.assertQueue(q);
    ch.consume(q, function(msg) {
      if (msg !== null) {
        console.log(msg.content.toString());
        ch.ack(msg);
      }
    });
  }
}
```

The returned message is `message-2`. When I checked our dashboard for the contents of the queue, there are no more messages left to consume.

Is this an expected behavior or should I put the each messages on a separate queue?

Thanks for the assist!

**Comment by cressie176** [comment]
According to the [channel api docs](http://www.squaremobius.net/amqp.node/channel_api.html) ```ch.consume``` set up a consumer with a callback to be invoked with each message. If you want to retrieve messages individually use ```ch.get```

**Comment by squaremo** [comment]
Nope, the expected behavior is that both `message-1` and `message-2` are printed to the console. So there is something else going on.

How do you know there are two messages, is that what the dashboard shows before you start the consumer? What happens if you use `basic.get` instead? Could there be another process taking messages?
